src/NuGet.Jobs.Common/JobBase.cs

```
@@ -30,14 +38,178 @@ protected JobBase(EventSource jobEventSource)
 
         protected ILogger Logger { get; private set; }
 
+        private Dictionary<string, ISqlConnectionFactory> SqlConnectionFactories { get; }
+
         public void SetLogger(ILoggerFactory loggerFactory, ILogger logger)
         {
             LoggerFactory = loggerFactory;
             Logger = logger;
         }
 
+        /// <summary>
+        /// Initialize the job, provided the service container and configuration.
+        /// </summary>
         public abstract void Init(IServiceContainer serviceContainer, IDictionary<string, string> jobArgsDictionary);
 
+        /// <summary>
+        /// Run the job.
+        /// </summary>
         public abstract Task Run();
+
+
+        /// <summary>
+        /// Test connection early to fail fast, and log connection diagnostics.
+        /// </summary>
+        private async Task TestConnection(string name, ISqlConnectionFactory connectionFactory)
```

I don't understand how this "fails fast". Isn't the failure swallowed into a log?

---

Yes, "warn fast" is more correct - just wanted a way to verify on startup whether the DB connections are configured correctly.

Maybe I should retry/throw in this case?

---

Updated to really fail fast. Will retry on job restart, and hopefully we can add heartbeat in future to detect startup failures.